doc/source/tutorial/tutorial_sessions.ipyml

```diff
@@ -8,14 +8,115 @@ cells:
+      
+     \* The *autocomplete* is the feature in which development tools try to predict the variable or function a user intends to enter after only a few characters have been typed (like word completion in cell phones).
+
+
+- markdown: |
+    ## Creating Sessions 
+
+
+- markdown: |
+    ### Session
+    
+    Create a session:
+
+
+- code: |
+    # define some scalars, axes and arrays
+    variant = 'baseline'
+    
+    country = Axis('country=Belgium,France,Germany')
+    gender = Axis('gender=Male,Female')
+    time = Axis('time=2013..2017')
+    
+    population = zeros([country, gender, time])
+    births = zeros([country, gender, time])
+    deaths = zeros([country, gender, time])
+
+
+- code: |
+    # create an empty session and objects one by one after
+    s = Session()
+    s.variant = variant
+    s.country = country
+    s.gender = gender
+    s.time = time
+    s.population = population
+    s.births = births
+    s.deaths = deaths
+    
+    print(s.summary())
+
+
+- code: |
+    # or create a session in one step by passing all objects to the constructor
+    s = Session(variant=variant, country=country, gender=gender, time=time, 
+                population=population, births=births, deaths=deaths)
+    
+    print(s.summary())
+
+
+- markdown: |
+    ### CheckedSession
+    
+    The syntax to define a checked-session is a bit specific:
+    
+    ```python
+    class MySession(CheckedSession):
+        # Variables can be declared in two ways:
+        # a) by specifying only the type of the variable (to be initialized later)
+        var1: Type
+        # b) by giving a default value
+        var2 = initialization value
+        # Additionally, axes and dtype of Array variables can be constrained using CheckedArray
+        arr1: CheckedArray([list, of, axes], dtype) = initialization value
+    ```
+    
+    Check the example below:
+
+
+- code: |
+    class Demography(CheckedSession):
+        # (convention is to declare parameters (read-only objects) in capital letters)
+        # Declare 'VARIANT' parameter as of type string.
+        # 'VARIANT' will be initialized when a 'Demography' session will be created
+        VARIANT: str
+        # declare variables of type Axis with an initialization value.
+        COUNTRY = Axis('country=Belgium,France,Germany')
+        GENDER = Axis('gender=Male,Female')
+        TIME = Axis('time=2013..2017')
+        # declare 'population' and 'births' arrays with an initialization value.
+        population = zeros([COUNTRY, GENDER, TIME], dtype=int)
+        births = zeros([COUNTRY, GENDER, TIME], dtype=int)
+        # declare 'deaths' with constrained axes and dtype.
+        # Its type (Array), axes and dtype are not modifiable.
+        # It will be initialized with 0
+        deaths: CheckedArray([COUNTRY, GENDER, TIME], int) = 0
```

Comment:
Doesn't CheckedArray also support axes **names** in addition to real Axis objects? If so, it would probably be a good idea to mention this. I suppose there are case where that can be useful (axes can vary or are defined later).

Reply:
No it doesn't.